com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs

```diff
@@ -153,7 +153,7 @@ bool StripUnusedPass(ShaderFeatures features, ShaderSnippetData snippetData)
 
         bool StripUnusedFeatures(ShaderFeatures features, Shader shader, ShaderSnippetData snippetData, ShaderCompilerData compilerData)
         {
-            bool stripDebugDisplayShaders = !Debug.isDebugBuild;
+            bool stripDebugDisplayShaders = !Debug.isDebugBuild || !UniversalRenderPipelineGlobalSettings.instance.supportRuntimeDebugDisplay;
```

Comment:
can URP Global Settings be null?

Reply:
Good point, the instance can indeed be null - I assumed the code would ensure it exists but looks like a lot of that code is editor-only, so there might be cases in player builds where it's hasn't been set. I'll add a check to be sure.

Reply:
Calling Ensure() here now (which asserts on failure to create) like HDRP does just to make sure.